src/processes.jl

```diff
@@ -44,22 +44,33 @@ Generate a single realization of spatial point process
 function rand_single end
 
 """
-    default_sampling_algorithm(p, g)
+    default_sampling_algorithm(p)
 
-Default sampling algorithm for spatial point process `p`
-on geometry `g`.
+Default sampling algorithm for spatial point process `p`.
 """
 function default_sampling_algorithm end
 
 # --------------------
 # SAMPLING ALGORITHMS
 # --------------------
 
-struct ProductSampling end
-struct ThinnedSampling end
 struct DiscretizedSampling end
 struct UnionSampling end
 
+"""
+    ThinnedSampling(param)
+
+Generate sample using Lewis-Shedler algorithm (1979) using `param`.
+It can be a real value to define the maximum value of the intensity
+function or an `NTuple` to define the dimensions of the
+`CartesianGrid` used to obtain the intensity maximum value.
+"""
+struct ThinnedSampling{T<:Union{Real,Dims}}
+  param::Union{Nothing,T}
+end
```

Comment:
This design isn't ideal. I think we can compute the lambda_max or any parameter inside of the method definition and leave the ThinnedSampling algorithm as an empty struct.